### PR TITLE
Feature/handle ia endpoint fail

### PR DESCRIPTION
**Pull request by hyzhak**
issue #82

- added special http error
- match http error to dedicated action handler

### CODE REVIEW

File: functions/tests/integration/alexa/dialog.yaml [hunk 1]
```diff
@@ -45,8 +45,18 @@
...
                 please try again later or try something else.'
+
+- scenario: "failed request for a collection from IA should gracefully fail in music query intent"
+  dialog:
+  - user: 'seventy eighties'
+    assistant: 'You''ve selected 78s.
+                What genre of music would you like to listen to?
+                Please select a genre like Jazz, Instrumental or Dance?'
+  - user: 'turbo folk music'
+    assistant: 'We experience some technical problems on IA server,
```
Comment (adaveinthelife) [inline functions/tests/integration/alexa/dialog.yaml]: Code looks great Eugene! I would change the last assistant line from `We experience some technical problems on IA server` to `We are currently experiencing some technical difficulties on the Archive server. Please try again later or try saying something else.` Sounds more natural!

File: functions/src/strings.js [hunk 2]
```diff
@@ -94,6 +94,13 @@ module.exports = {
       suggestions: ['reset'],
     },
 
+    /**
+     * we got 4xx, 5xx response from a server
+     */
+    httpRequestError: {
+      speech: 'We experience some technical problems on IA server, please try again later or try something else.',
```
Comment (adaveinthelife) [inline functions/src/strings.js]: Change this line to `We are currently experiencing some technical difficulties on the Archive server. Please try again later or try saying something else.`.

File: functions/tests/platform/assistant/handler.spec.js [hunk 3]
```diff
@@ -86,6 +89,28 @@ describe('platform', () => {
...
+        handler(buildIntentRequest({
+          action: 'on-action',
+          lastSeen: null,
+        }), res);
+
+        return wait()
+          .then(() => {
+            expect(res.speech()).to.includes(
+              'We experience some technical problems on IA server, ' +
+              'please try again later or try something else.'
```
Comment (adaveinthelife) [inline functions/tests/platform/assistant/handler.spec.js]: Change this to `We are currently experiencing some technical difficulties on the Archive server. Please try again later or try saying something else.`

File: functions/src/strings.js [hunk 4]
```diff
@@ -151,13 +151,20 @@ module.exports = {
...
+       * - doesn't match to empty suggestions
        */
       repair: {
         speech: [
           `I don’t have anything for {{year}}. Try {{suggestions.0}}, for example.`,
           `I don't have {{creator}} albums for {{year}}. Try {{suggestions.0}}, for example.`,
           `I don't have any albums for {{year}}. Try {{suggestions.0}}, for example.`,
           `I don't have that. Try {{suggestions.0}}, for example.`,
+          `I don't have {{subject}} of {{year}}. Try {{suggestions.0}}, for example.`,
+          `I don't have {{subject}} of {{year}}. Maybe you would like to listen something else?`,
```
Comment (adaveinthelife) [inline functions/src/strings.js]: The grammar here is a little clunky. I would change it from `I don't have {{subject}} of {{year}}` to `I don't have {{subject}} for {{year}}`

Comment (adaveinthelife) [reply functions/src/strings.js]: Same thing goes for the previous instance in the line above this one.

File: functions/tests/integration/alexa/dialog.yaml [hunk 5]
```diff
@@ -60,3 +60,18 @@
...
+  - user: '1990 year'
+  - user: 'play folk'
+    assistant: "I don't have folk of 1990. Maybe you would like to listen something else?"
+
+- scenario: 'handle empty playlist in one go intent'
+  dialog:
+  - user: '1910 year'
+  - user: 'play rock'
+    assistant: "I don't have rock of 1910. Try USA 1910, for example."
+# TODO:
```
Comment (adaveinthelife) [inline functions/tests/integration/alexa/dialog.yaml]: If you update the dialog - you'll want to fix these tests. Change `of {year}` to `for {year}`.

**Comment by hyzhak** [comment]
@adaveinthelife thanks for you code review! it seems I have pushed commit from another branch to this one, I have just rebased this one to its branch.

**Pull request merged**